src/api/ZoweExplorerZosmfApi.ts

```diff
@@ -29,24 +30,67 @@ class ZosmfApiCommon implements ZoweExplorerApi.ICommon {
     }
 
     private session: Session;
+    private log: Logger;
     constructor(public profile?: IProfileLoaded) {
     }
 
     public getProfileTypeName(): string {
         return ZosmfUssApi.getProfileTypeName();
     }
 
+    public getSessionFromCommandArgument(cmdArgs: ICommandArguments): Session{
+        return zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
+    }
+
     public getSession(profile?: IProfileLoaded): Session {
         if (!this.session) {
-            this.session = zowe.ZosmfSession.createBasicZosmfSession((profile||this.profile).profile);
+            try {
+                if (!this.profile.profile.tokenValue) {
+                    this.session = zowe.ZosmfSession.createBasicZosmfSession((profile||this.profile).profile);
```

Comment:
Should call `this.getSessionFromCommandArgument()`

Reply:
this line if for basic zosmf session so think it's ok as is